### DISCUSSION TITLE
How to invoke PyNifly import& export through python script correctly?

### DISCUSSION THREAD

**Discussion by hxak016** [BadDogSkyrim/PyNifly#252] [Q&A]
First of all I just have to say a huge thank you for all the amazing work you’ve done on PyNifly! Seriously, without this addon, so many things would just not be possible. 

The tools you've created have made such a huge impact, and the way you’ve made Python and NIF file manipulation work together so smoothly is like, total magic for modders!  It’s so obvious you really get both the technical side and what the modding community needs, and your contributions are totally changing the game in how mods are made and perfected.

After using PyNifly for a while, I want to try using it through script/batch for some repetitive work (e.g. reshape all hair and hair line nifs for a certain custom race). However, without enough scripting knowledge, I don't know how to invoke import& export correctly. Below is what I've managed to come up with:

```
import bpy
import os

filepath = r"D:\Desktop\YuiTest\awesomeTest.nif"

bpy.ops.import_scene.pynifly(filepath='', filter_glob='*.nif', files=<bpy_collection[0], IMPORT_SCENE_OT_pynifly.files>, do_create_bones=True, use_blender_xf=False, do_rename_bones=True, do_import_animations=True, do_import_collisions=True, do_import_tris=True, rename_bones_niftools=False, do_import_shapes=True, do_apply_skinning=True, do_import_pose=False, do_create_collections=False, reference_skel='')
```

It would be incredibly helpful if you could give me a heads up on how to use PyNifly through script! 

Once again, your dedication to making the modding community better is honestly so inspiring!  

Keep being awesome!

**Comment by BadDogSkyrim** [comment]
I think there's a file called _blender_tests.py_ in the kit I ship. Open up the zip and see. Those are the functional tests I use, and they're all examples of scripting pynifly, with more or less manipulation of the mesh in blender depending on the test.

**Reply by BadDogSkyrim** [reply]
Thanks for the good words, btw.
